eZ/Publish/Core/FieldType/Keyword/KeywordStorage/Gateway/LegacyStorage.php

```diff
@@ -378,4 +387,54 @@ protected function deleteOrphanedKeywords()
 
         $deleteQuery->prepare()->execute();
     }
+
+    /**
+     * @param \eZ\Publish\Core\Persistence\Database\DatabaseHandler $dbHandler
+     * @param mixed $fieldId
+     * @param int $versionNo
+     * @return bool
+     */
+    private function publishedVersionForFieldExists(
+        DatabaseHandler $dbHandler,
+        $fieldId,
+        $versionNo
+    ) {
+        $selectQuery = $dbHandler->createSelectQuery();
+        $selectQuery->select($dbHandler->quoteColumn('id', 'ezcontentobject_attribute'))
+            ->from(
+                $dbHandler->quoteTable('ezcontentobject_attribute')
+            )
+            ->innerJoin(
+                $dbHandler->quoteTable('ezcontentobject_version'),
+                $dbHandler->quoteColumn('contentobject_id', 'ezcontentobject_attribute'),
+                $dbHandler->quoteColumn('contentobject_id', 'ezcontentobject_version')
+            )
+            ->where(
+                $selectQuery->expr->lAnd(
+                    $selectQuery->expr->eq(
+                        $dbHandler->quoteColumn('id', 'ezcontentobject_attribute'),
+                        $selectQuery->bindValue($fieldId, null, \PDO::PARAM_INT)
+                    ),
+                    $selectQuery->expr->neq(
+                        $dbHandler->quoteColumn('version', 'ezcontentobject_attribute'),
+                        $selectQuery->bindValue($versionNo, null, \PDO::PARAM_INT)
```

Comment:
Right, but I think if I pass `VersionInfo` instead of `ContentInfo` I don't even need to pass `$versionNo` separately. Changed in a669b2c.

A little bit explanation on this change: if current version (`VersionInfo`) is not published, but underlying `ContentInfo` is published this means basically what query checked.
 Both integration tests and manual web app tests seems to confirm this check is enough.

*I must say, I like querying database too much... Sorry :-)*